File: app/src/main/java/com/sd/style/module/home/v/fragment/HomeFragment.java
package com.sd.style.module.home.v.fragment;

import android.widget.Button;

import com.sd.style.R;
import com.sd.style.common.base.BaseFragment;
import com.sd.style.common.base.BasePresenter;
import com.sd.style.module.TestActivity;

import butterknife.BindView;

/**
 * Author: HeLei on 2017/10/16 18:25
 */

public class HomeFragment extends BaseFragment {

    @BindView(R.id.finger_test)
    Button finger_test;

    public static HomeFragment newInstance() {
        return new HomeFragment();
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_home;
    }

    @Override
    protected void initView() {

    }

    @Override
    protected void bindData() {
        finger_test.setOnClickListener(v ->
                TestActivity.showActivity(mContext)
        );
    }

    @Override
    protected BasePresenter initPresenter() {
        return null;
    }
}

File: app/src/main/java/com/sd/style/module/home/v/fragment/HomeFragment.kt
package com.sd.style.module.home.v.fragment


import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import com.sd.style.R
import com.sd.style.common.base.BaseFragment
import com.sd.style.common.base.BasePresenter
import kotlinx.android.synthetic.main.base_title.*


/**
 * Author: HeLei on 2017/10/16 18:25
 */

class HomeFragment : BaseFragment() {

    override fun getLayoutId(): Int {
        return R.layout.fragment_home
    }

    override fun initView() {
        //强转为AppCompatActivity
        setHasOptionsMenu(true)
        initToolbar()
        changeTitleHeight()
    }

    private fun initToolbar() : Toolbar{
        val activity :AppCompatActivity? = mContext as AppCompatActivity
        activity?.setSupportActionBar(base_toolbar)
        base_toolbar.title = "首页"
        base_toolbar.setNavigationIcon(R.drawable.title_back)
        return base_toolbar
    }

    override fun bindData() {
    }

    override fun initPresenter(): BasePresenter<*>? {
        return null
    }

    companion object {

        fun newInstance(): HomeFragment {
            return HomeFragment()
        }
    }
}
